streamline/base.py

```diff
@@ -136,6 +141,26 @@ class method.
         """
         return cls.name or cls.get_generic_name()
 
+    @staticmethod
```

Comment:
No special opinion, just seemed natural.. as long as `before_hooks` / `after_hooks` aren't redefined on a subclass, classmethods should be just as reliable